MapboxNavigation/PollyVoiceController.swift

```diff
@@ -104,24 +95,35 @@ public class PollyVoiceController: RouteVoiceController {
                 return nil
             }
             
-            strongSelf.handle(awsTask)
+            strongSelf.handle(awsTask, for: routeProgress, userDistance: userDistance)
             
             return nil
         }
     }
     
-    func callSuperSpeak(_ text: String, error: String) {
-        super.speak(fallbackText, error: error)
+    /**
+     Speak using a fallback mechanism that doesn’t involve Polly.
+     
+     This method exists because it isn’t possible to refer to super inside a closure.
+     */
+    func speakWithoutPolly(for routeProgress: RouteProgress, userDistance: CLLocationDistance) {
+        super.speak(for: routeProgress, userDistance: userDistance)
     }
     
-    func handle(_ awsTask: AWSTask<NSURL>) {
-        guard awsTask.error == nil else {
-            super.speak(fallbackText, error: awsTask.error!.localizedDescription)
+    func handle(_ awsTask: AWSTask<NSURL>, for routeProgress: RouteProgress, userDistance: CLLocationDistance) {
+        if let error = awsTask.error {
+            print(error)
+            speakWithoutPolly(for: routeProgress, userDistance: userDistance)
+            return
+        }
+        
+        guard !awsTask.isCancelled else {
             return
         }
         
         guard let url = awsTask.result else {
-            super.speak(fallbackText, error: "No polly response")
+            print("No polly response")
+            speakWithoutPolly(for: routeProgress, userDistance: userDistance)
             return
         }
         
```

Comment:
Should we cancel the task before creating a new one?

Reply:
Good idea, but I think we should address this issue in #661. Once that PR lands, I’ll rebase and do likewise here.